From: Prarit Bhargava <prarit@redhat.com>
Date: Tue, 15 Jul 2008 19:57:55 -0400
Subject: [x86] don't call MP_processor_info for disabled cpu
Message-id: 20080715235717.12212.50595.sendpatchset@prarit.bos.redhat.com
O-Subject: [RHEL5 PATCH]: don't call MP_processor_info for disabled cpu (32 bit)
Bugzilla: 455425
RH-Acked-by: Brian Maly <bmaly@redhat.com>
RH-Acked-by: Rik van Riel <riel@redhat.com>

Backport of 987dd2d4d465e80e00d6a0b16787a78aa75dd66a

x86: don't call MP_processor_info for disabled cpu

    Signed-off-by: Alexey Starikovskiy <astarikovskiy@suse.de>
    Signed-off-by: Ingo Molnar <mingo@elte.hu>

Successfully tested by me.

Resolves BZ 455425

diff --git a/arch/i386/kernel/mpparse.c b/arch/i386/kernel/mpparse.c
index 8aca0f6..40f4daf 100644
--- a/arch/i386/kernel/mpparse.c
+++ b/arch/i386/kernel/mpparse.c
@@ -842,7 +842,10 @@ void __cpuinit mp_register_lapic (
 {
 	struct mpc_config_processor processor;
 	int			boot_cpu = 0;
-	
+
+	if (!enabled)
+		return;
+
 	if (MAX_APICS - id <= 0) {
 		printk(KERN_WARNING "Processor #%d invalid (max %d)\n",
 			id, MAX_APICS);
